FunctionalTableData/TableView/FunctionalTableData.swift

```diff
@@ -400,18 +400,31 @@ public class FunctionalTableData: NSObject {
 		CATransaction.setCompletionBlock {
 			completion?()
 		}
-		
-		tableView.beginUpdates()
+
 		// #4629 - There is an issue where on some occasions calling beginUpdates() will cause a heightForRowAtIndexPath() call to be made. If the sections have been changed already we may no longer find the cells
 		// in the model causing a crash. To prevent this from happening, only load the new model AFTER beginUpdates() has run
-		sections = localSections
-		applyTableSectionChanges(changes)
-		tableView.endUpdates()
+		if #available(iOS 11.0, *) {
```

Comment:
Could be worth separating these branches into 2 inline functions so we can handle the completion handler differently for both and don't have as many `if #available` checks.

Does the `performBatchUpdates` need to be within the `CATransaction`s?

Reply:
I am with you on this Caio